nexus/db-queries/src/db/datastore/sled.rs

```diff
@@ -564,14 +736,370 @@ impl DataStore {
                 resources.clone(),
             );
 
-            // Try to INSERT the record. If this is still a valid target, we'll
-            // use it. If it isn't a valid target, we'll shrink the set of
-            // viable sled targets and try again.
-            let rows_inserted = sled_insert_resource_query(&resource)
+            if local_storage_allocation_required.is_empty() {
+                info!(
+                    &log,
+                    "attempting to insert sled resource record";
+                    "local storage required" => "no",
```

Comment:
turbo nit: why not `false`?
also, i think that since this gets turned into a JSON field name, it should maybe be `local_storage_required`
```suggestion
                    "local_storage_required" => false,
```